### DISCUSSION TITLE
Wednesday, May 18, 2022 @ 5:00pm at the Saratoga Springs Library

### DISCUSSION THREAD

**Discussion by blainsmith** [agnostechvalley/agnostechvalley.github.io#18] [Announcements]
I scheduled a meetup location! [Saratoga Springs Library, Glasby Room](https://www.sspl.org/services/meeting_rooms/): 5:00pm - 8:00pm Wednesday, May 18, 2022. Once I get confirmation we have the room from the coordinator I will let everyone know, but wanted to give you all a heads up. It is a 13 person room with AV equipment, etc.

Presenters:

- `AF_PACKET`, raw packets, and timestamps on Linux with Blain Smith
- Using Tilt with different K8s environments with Chief Najewicz

If you would like to give a demo, talk, etc just reply in the comments and we'll keep a running list of folks.

**Comment by blainsmith** [comment]
I can demo some of the stuff I've been doing with sending/receiving raw packets in/out of specific network interfaces using Linux's `AF_PACKET` and BPF filtering with raw low level sockets. This requires you construct each layer of a network packet (Layer 1 - 4) in order to make things work. It will also be a preview of the open source package that will let you do this easily that we plan to release.

![Internet Model](https://linuxhint.com/wp-content/uploads/2021/10/TCPIP-layers-and-their-functions-1.jpg)

**Comment by chiefy** [comment]
I can do a demo of [tilt](https://tilt.dev/) including some overview on how to use it with different k8s deployments (k3s, EKS, GKE, etc.)
